dependencies.yaml

```diff
@@ -526,7 +526,15 @@ dependencies:
           - statsmodels
           - umap-learn==0.5.6
           - pynndescent
-          - setuptools  # Needed on Python 3.12 for dask-glm, which requires pkg_resources but Python 3.12 doesn't have setuptools by default
```

Comment:
`dask-glm` was removed by this PR https://github.com/rapidsai/cuml/pull/6028 so this is now unnecessary